src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/Transport/TTransport.cs

```diff
@@ -91,17 +91,17 @@ protected static void ValidateBufferArgs(byte[] buffer, int offset, int length)
 #if DEBUG // let it fail with OutOfRange in RELEASE mode
             if (offset < 0)
             {
-                throw new ArgumentOutOfRangeException(nameof(offset), "Buffer offset must be >= 0");
+                throw new ArgumentOutOfRangeException(nameof(offset), offset, $"Buffer offset must be >= 0, but was: {offset}");
```

Comment:
General question - do we want to put "but was: {offset}" in the message or not.

I suggest that we don't, since the actual value is already provided as the 2nd argument, and is handled in a systematic way:

```csharp
throw new ArgumentOutOfRangeException("arg1", 1, "invalid argument");
```

The actual value is provided by the exception:

```text
Unhandled exception. System.ArgumentOutOfRangeException: invalid argument (Parameter 'arg1')
Actual value was 1.
   at Program.Main() in Program.cs:line 5
```

I would also suggest that the exception message (3rd argument) to be vanilla string, instead of an interpolation. The main benefit is the ability to group exception messages without having to do string parsing or pattern matching. There are other benefits such like avoiding one heap allocation, but that's probably okay.

Reply:
Should I remove all Value's in description?

Reply:
I would vote for YES - I think the 3rd argument should be a plain string (doesn't require extra formatting/allocation/parsing).
Other maintainers/approvers might have different opinion / personal taste though 😃

Reply:
The main reason I put it in, if someone is only logging the Exception message. That way they would have enough information to solve the issue quicker without needing to debug (or at least that was the plan.)

Reply:
I can see other places in code there is stuff like: `throw new ArgumentException($"{nameof(processors)} collection is empty");` and `throw new InvalidOperationException($"The context slot {slotName} is already registered.");`

Reply:
There are similar debates around structured logging vs. unstructured logging.

My two cents:

1. If the goal is to encourage folks to put all the info in the exception message, there shouldn't be `ctor(name, value, message)` at all. So I guess .NET folks invented this ctor to discourage people from putting the actual value in the message?

2. In general the industry is moving towards structured logging, and in OpenTelemetry folks have the same push:

_The string is a stark data structure and everywhere it is passed there is much
duplication of process. It is a perfect vehicle for hiding information._

-- Alan J. Perlis

Reply:
> {nameof(processors)} collection is empty

And most likely if you run `git blame` both would point to me 😄

Reply:
I have removed all the values and ended messages with "."